README.md

```diff
@@ -0,0 +1,60 @@
+# tool-upgrade-go
+
+A library for golang that helps upgrade tools to newer versions with ease.
+
+## Usage
+
+### Repository structure
+
+```
+# over-simplified example of directory and file structure required
+# mycompany.com web server directory or artifact repository
```

Comment:
maybe it's implied, but do it need to state that the artifact repository needs to be http accessible?

Reply:
I think HTTP(S) is implied by "web server", would you agree? How would you clarify it?